### PR TITLE
[FLINK-23028][docs] Improve documentation for pages of SQL.

### PR DESCRIPTION
**Pull request by RocMarshal**
## What is the purpose of the change

Improve documentation for pages of SQL. 

## Brief change log

Correct wrong style in https://ci.apache.org/projects/flink/flink-docs-master/zh/docs/dev/table/sql/create/#create-function   section.

Correct wrong style in https://ci.apache.org/projects/flink/flink-docs-master/zh/docs/dev/table/sql/drop/#drop-function section.

Correct wrong style in https://ci.apache.org/projects/flink/flink-docs-master/zh/docs/dev/table/sql/alter/#alter-function section.

Add the description about drop catalog in the page of  https://ci.apache.org/projects/flink/flink-docs-master/docs/dev/table/sql/drop/ .

Correct 'catloag' of  https://ci.apache.org/projects/flink/flink-docs-master/zh/docs/dev/table/sql/use/#use-catloag

*A pure documentation work.*

## Verifying this change

Correct wrong style in https://ci.apache.org/projects/flink/flink-docs-master/zh/docs/dev/table/sql/create/#create-function   section.

Correct wrong style in https://ci.apache.org/projects/flink/flink-docs-master/zh/docs/dev/table/sql/drop/#drop-function section.

Correct wrong style in https://ci.apache.org/projects/flink/flink-docs-master/zh/docs/dev/table/sql/alter/#alter-function section.

Add the description about drop catalog in the page of  https://ci.apache.org/projects/flink/flink-docs-master/docs/dev/table/sql/drop/ .

Correct 'catloag' of  https://ci.apache.org/projects/flink/flink-docs-master/zh/docs/dev/table/sql/use/#use-catloag

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by RocMarshal** [comment]
@xccui  Could you help me to review this PR?Thanks.

**Comment by RocMarshal** [comment]
@klion26 @xccui Could you help me to review this PR? Thank you.

**Comment by RocMarshal** [comment]
@dianfu  Could you help me to review this PR? Thank you.

**Comment by RocMarshal** [comment]
@alpinegizmo @dianfu Could you help me to review this PR? Thank you.

File: docs/content/docs/dev/table/sql/drop.md [hunk 1]
```diff
@@ -26,10 +26,11 @@ under the License.
 
 # DROP Statements
 
-DROP statements are used to remove a registered table/view/function from current or specified [Catalog]({{< ref "docs/dev/table/catalogs" >}}).
+DROP statements are used to remove a catalog with given catalog name or used to remove a registered table/view/function from current or specified [Catalog]({{< ref "docs/dev/table/catalogs" >}}).
```
Comment (alpinegizmo) [inline docs/content/docs/dev/table/sql/drop.md:29]:
```suggestion
DROP statements are used to remove a catalog with the given catalog name or to remove a registered table/view/function from the current or specified [Catalog]({{< ref "docs/dev/table/catalogs" >}}).
```

**Review by alpinegizmo — COMMENTED** [review]
I've left one suggestion for improving the English.

**Comment by RocMarshal** [comment]
@JingsongLi  Could you help me to review the part of Chinese in this PR? Thank you.

**Comment by RocMarshal** [comment]
@flinkbot run azure

**Review by Airblader — CHANGES_REQUESTED** [review]
Thanks for your PR! Could you please double-check all of these also in the English versions of those pages and fix it there as well? Thank you!

**Comment by RocMarshal** [comment]
@Airblader Thank you for your suggestions. I did some checking based on your comments, but it seems that the problem described in JIRA has been fixed now.

**Comment by RocMarshal** [comment]
@gaoyunhaii Could you help me to review this PR? Thank you.

File: docs/content.zh/docs/dev/table/sql/drop.md [hunk 2]
```diff
@@ -146,6 +147,18 @@ Flink SQL> SHOW TABLES;
 {{< /tab >}}
 {{< /tabs >}}
 
+## DROP CATALOG
+
+```sql
+DROP CATALOG [IF EXISTS] catalog_name
+```
+
+删除给定的 catalog name 的 catalog。
```
Comment (gaoyunhaii) [inline docs/content.zh/docs/dev/table/sql/drop.md:156]: 删除给定名子的 Catalog ?

File: docs/content.zh/docs/dev/table/sql/drop.md [hunk 3]
```diff
@@ -146,6 +147,18 @@ Flink SQL> SHOW TABLES;
...
+
+**IF EXISTS**
+
+如果目标 catalog 不存在，则不会生效。
```
Comment (gaoyunhaii) [inline docs/content.zh/docs/dev/table/sql/drop.md:160]: 则不会生效 => 不会执行任何操作？

**Comment by gaoyunhaii** [comment]
Hi @RocMarshal Very thanks for the contribution ! I checked the style fix and it should indeed works. I left two small comments regarding the Chinese description. The other part LGTM.

**Comment by RocMarshal** [comment]
@gaoyunhaii Thank you for your review. I made some changes based on your comments. 
PTAL.

**Comment by gaoyunhaii** [comment]
Thanks @RocMarshal for the update and LGTM now, and we would still need a committer to finally confirm and merge the PR~

**Comment by RocMarshal** [comment]
Hi, @alpinegizmo @wuchong ,Could you help me to confirm and merge the PR ? Thank you so much for your help.

**Review by JingsongLi — APPROVED** [review]
Thanks @RocMarshal ! The PR looks good to me!

**Review by JingsongLi — COMMENTED** [review]
And thanks @alpinegizmo and @gaoyunhaii for the review. I will merge this PR~

**Pull request merged**